### PR TITLE
Query database to get lists of channels, posts, comments for indexing

### LINKED ISSUE #1368: Missing posts in the elasticsearch index
We need to change our reindexing code to reference our database models to get the post and channel information, rather than asking reddit for it since reddit cuts off posts at about 1000 posts per channel

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1368 

#### What's this PR do?
- Queries the database instead of reddit to get the channels, posts, and comments to index.
- Adds factories for Django models (Channel, Post, Comment) and refactors existing factory class names for the reddit equivalents to differentiate between them.

#### How should this be manually tested?
- Run `docker-compose run web python manage.py recreate_index`
- Query Elasticsearch to verify that all posts and comments were indexed.

### CODE REVIEW

File: search/serializers.py [hunk 1]
```diff
@@ -162,14 +171,16 @@ def _serialize_comment_tree_for_bulk(comments):
     Serialize an iterable of Comment and their replies for a bulk API request
 
     Args:
-        comments (iterable of Comment): An iterable of comments
+        comments (iterable of str): An iterable of comment ids
```
Comment (noisecapella) [inline search/serializers.py]: Can we change `comments` to `comment_ids`?

File: search/serializers.py [hunk 2]
```diff
@@ -162,14 +171,16 @@ def _serialize_comment_tree_for_bulk(comments):
     Serialize an iterable of Comment and their replies for a bulk API request
```
Comment (noisecapella) [inline search/serializers.py]: Can you update this? In particular we should remove `and their replies`

File: search/tasks.py [hunk 3]
```diff
@@ -44,6 +43,10 @@ def wrap_retry_exception(*exception_classes):
     try:
         yield
     except Exception as ex:
+        if isinstance(ex, NotFound):
+            # No corresponding reddit post/comment found for django model object.  Log error and continue indexing.
+            log.exception(str(ex))
```
Comment (noisecapella) [inline search/tasks.py]: You can omit the `str(ex)` here, the whole stack trace will be logged automatically

Comment (mbertrand) [reply search/tasks.py]: ✔️

File: search/serializers.py (hunk 1 above)
Comment (mbertrand) [reply search/serializers.py]: ✔️

File: search/tasks.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def wrap_retry_exception(*exception_classes):
     except Exception as ex:
         if isinstance(ex, NotFound):
             # No corresponding reddit post/comment found for django model object.  Log error and continue indexing.
-            log.exception(str(ex))
+            log.exception()
```
Comment (noisecapella) [inline search/tasks.py]: Actually it looks like this requires an argument. Can you add a brief message? The exception stacktrace will be logged before it automatically

Comment (mbertrand) [reply search/tasks.py]: ✔️

**Pull request merged**